python/lsst/ts/ofc/controllers/pid_controller.py

```diff
@@ -0,0 +1,98 @@
+# This file is part of ts_ofc.
+#
+# Developed for Vera Rubin Observatory.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import logging
+
+import numpy as np
+
+from .. import OFCData
+from . import BaseController
+
+__all__ = ["PIDController"]
+
+
+class PIDController(BaseController):
+    """PID controller.
+
+    Parameters
+    ----------
+    kp : `float`
+        Proportional gain.
+    ki : `float`
+        Integral gain.
+    kd : `float`
+        Derivative gain.
+    setpoint : `float`, optional
+        Setpoint for the controller.
```

Comment:
That's because those attributes are set in the parent class. Should I then remove them from this docstring, even though they are also attributes of the PIDController?

Reply:
If the `Child.__init__()` and `Parent.__init__()` have the same signatures and class attributes, you can remove the doc string of `__init__()` from the Child. Otherwise, you may need to copy what you have from the Parent.

As an example, you can do the following experiment:

```python
class Parent()
    """Doc string of parent"""
    def __init__():
        self.a = "foo"

class ChildA(Parent)
    """Doc string of parent"""
    def __init__():
        self.b = "bar"

class ChildB(Parent)
    def function1:
        pass

child_a = ChildA()
child_b = ChildB()

# Do the help for each child to check the doc string and attribute
help(child_a)
help(child_b)
```

And sometimes, based on the condition, we will use the `super()` function in child as well, which depends on the use case.

Reply:
Therefore, in your case, for the `PIDController`, you can remove the function of `__init__()` because it does not add any new logic from the parent. You can try this to verify it.

In the `OICController`, you would need to rewrite all the details of doc string because you overwrote the function (`__init()__`) from the parent with the new attributes.